### PR TITLE
fixes #18864 - option to clear subscriptions first

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

**Comment by thomasmckay** [comment]
@akofink - good for review

File: lib/hammer_cli_csv/content_hosts.rb [hunk 1]
```diff
@@ -348,12 +351,12 @@ def update_subscriptions(host, line, remove_existing)
...
+            subs = [{:id => existing_subscription['id'], :quantity => existing_subscription['quantity_consumed']}]
+            @api.resource(:host_subscriptions).call(:remove_subscriptions, {
+              'host_id' => host['id'],
+              'subscriptions' => subs
+            })
           end
-          @api.resource(:host_subscriptions).call(:remove_subscriptions, {
-            'host_id' => host['id'],
-            'subscriptions' => existing_subscriptions
-          })
```
Comment (thomasmckay) [inline lib/hammer_cli_csv/content_hosts.rb]: There was a race condition in candlepin, solved but safest to support older versions of foreman.
https://github.com/Katello/katello/pull/6722

**Comment by thomasmckay** [comment]
I will add verbose, that is a good idea. Let me check out the mutually exclusive options too.

**Comment by thomasmckay** [comment]
@akofink - updated to add message when clearing subs and check for use of --clear-subscriptions during export

**Comment by akofink** [comment]
This seems strange that the subscriptions are cleared after "updating subscriptions" for each host. Is this expected behavior? Also, should "clearing existing subscriptions" and "attaching" each be on a new line?

```sh
$ hammer csv content-hosts --verbose --itemized-subscriptions --file /tmp/content_host_itemized_subs.csv --clear-subscriptions
Updating subscriptions for content host 'akofink-desktop.usersys.redhat.com'...done
Updating subscriptions for content host 'katello-client.example.com'...done
Updating subscriptions for content host 'rhel1.example.com'...clearing existing subscriptions... attaching 'Employee SKU'...done
```

**Comment by thomasmckay** [comment]
@akofink - going to roll with this as is. The verbose message output has had some debate in the past but keeping a row of csv limited to a single line of output is the pattern to date.

**Comment by akofink** [comment]
Since my comments are not pertinent/you're fine without those changes, ACK!

**Pull request merged**